crates/router/src/db/payment_attempt.rs

```diff
@@ -798,6 +829,27 @@ mod storage {
                 }
             }
         }
+
+        async fn get_payment_filters(
+            &self,
+            pi: &[storage_models::payment_intent::PaymentIntent],
+            merchant_id: &str,
+            storage_scheme: enums::MerchantStorageScheme,
+        ) -> CustomResult<storage_models::payment_attempt::PaymentListFilters, errors::StorageError>
+        {
+            let database_call = || async {
+                let conn = connection::pg_connection_read(self).await?;
+                PaymentAttempt::get_payment_filters(&conn, pi, merchant_id)
+                    .await
+                    .map_err(Into::into)
+                    .into_report()
+            };
```

Comment:
why do we need to do this? wrap the db call in an async block. Will this be used in the `RedisKv` branch?